hack/jenkins/upload-to-gcs.sh

```diff
@@ -115,11 +121,26 @@ function upload_artifacts_and_build_result() {
       echo "Uploading build log"
       gsutil -q cp -Z -a "${gcs_acl}" "${WORKSPACE}/build-log.txt" "${gcs_build_path}"
     fi
+
+    # For pull jobs, keep a canonical ordering for tools that want to examine
+    # the output.
+    echo "aoeu debug: ${gcs_indirect_path}"
```

Comment:
yeah, it always downloads upload-to-gcs.sh from HEAD: https://github.com/kubernetes/test-infra/blob/b12293e6c4a49a968705775b06656d452f6db9e0/jenkins/job-configs/kubernetes-jenkins-pull/global.yaml#L7-L13

I think we discussed changing this to download only if necessary. cc @spxtr